src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.spring.security;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import java.util.Optional;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    @Autowired
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(username);
+        User user = optionalUser.orElseThrow(() ->
+                new RuntimeException("Can't find user by email: " + username));
+        UserBuilder userBuilder = withUsername(username);
+        userBuilder.password(user.getPassword());
+        userBuilder.authorities(user.getRoles()
+                .stream()
+                .map(r -> r.getRoleName().name())
+                .toArray(String[]::new));
```

Comment:
have your checked if your endoints are properly secured with this solution? Probably you need to add 'ROLE_' prefix to each role name here